### ISSUE TITLE
Style/RedundantBegin – false positive when calling a method on the block `end`

### ISSUE THREAD

**Issue by bquorning** [rubocop-hq/rubocop#8869]
In v0.93.0, `Style/RedundantBegin` started taking an offense with this code:

```rb
# frozen_string_literal: true

@memo ||= begin
  this
  that
end.foo do
  other_thing
end
```

--------

## Expected behavior

No offenses.

## Actual behavior

```
test.rb:3:11: C: Style/RedundantBegin: Redundant begin block detected.
@memo ||= begin
          ^^^^^
```

## Steps to reproduce the problem

On the RuboCop repository:
```
git checkout v0.93.0 && \
echo '# frozen_string_literal: true

@memo ||= begin
  this
  that
end.foo do
  other_thing
end' > test.rb && \
./exe/rubocop test.rb
```

## RuboCop version

```
$ ./exe/rubocop -V
0.93.0 (using Parser 2.7.1.5, rubocop-ast 0.7.1, running on ruby 2.6.6 x86_64-darwin19)
```
